.github/workflows/publish.yml

```diff
@@ -0,0 +1,50 @@
+name: Publish
+on:
+  release:
+    types:
+      - published
+
+env:
+  # mark:automatic-version
+  VERSION: "1.20231001.0"
+
+jobs:
+  push-release-images:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        include:
+          - image_name: relay
+          - image_name: gateway
+    permissions:
+      # Needed to upload artifacts to a release
+      packages: write
+      # Needed to login to GCP
+      id-token: write
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/gcp-docker-login
+        id: login
+        with:
+          project: firezone-staging
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{github.actor}}
+          password: ${{secrets.GITHUB_TOKEN}}
+      - name: Pull and push
+        env:
+          SRC_IMAGE: ${{ steps.login.outputs.registry }}/firezone/${{ matrix.image_name }}:${{ github.sha }}
+          DST_IMAGES: |
+            ghcr.io/firezone/${{ matrix.image_name }}:${{ github.sha }}
```

Comment:
We need to create prod env and bucket first.

Reply:
These are going to Github Container Registry so they're synchronized with our `/releases` page and changelogs:

![](https://user-images.githubusercontent.com/167144/275645757-36d89150-abbb-46d3-88fd-3a771b855875.png)

Reply:
Can you please leave a TODO to still push copies of all that to the Google Artifact Registry? We need it for two reasons: 
1. Backup for all our images in case GitHub goes down;
2. We will use them in our production - faster to pull, less traffic, and better control (eg. tag immutability).

Reply:
Yeah, I'll make sure to pull the staging `github.sha`'s down here instead of the `tag`'s so that we have the same images published that we tested on staging.

Reply:
Wait a minute -- this is already using the same `github.sha` as the staging images. Publishing a release doesn't create a new commit, it just tags the current drafted release (i.e. tip of `main`). So `github.sha` in this workflow and `github.sha` in `cd.yml` will be identical. This means the staging images are already in Google Artifact Registry -- we just need to promote them to production (`firezone-production` project?)